### PR TITLE
Compression

### PR DESCRIPTION
**Pull request by willmcgugan**
**What this PR solves**

- Adds per-message compression, according to https://tools.ietf.org/html/rfc7692
- Refactor frame parser
- Fixes tests which had server frames including a mask (not permitted by spec)
- Passes the autobahn test suite with 100% of the tests (nothing left unimplemented)

**Review**

- [X] Ready for review

### CODE REVIEW

File: lomond/constants.py [hunk 1]
```diff
@@ -22,3 +22,7 @@
     platform.release(),
     platform.platform()
 )
+
+SUPPORTED_COMRESSION = [
```
Comment (toudi) [inline lomond/constants.py]: Typo ;) Also, it seems that this constant is not used

**Comment by willmcgugan** [comment]
Patience guys. :-)
screen shot 2018-06-18 at 19 41 01

**Comment by toudi** [comment]
touche! This checkbox must have slipped trough my eyes :)

**Comment by willmcgugan** [comment]
@barfle @toudi Ready for a proper view now ;-)

**Comment by barfle** [comment]
Cool :-)

**Pull request merged**